addons/web/static/src/less/list_view.less

```diff
@@ -6,6 +6,14 @@
     background-color: @odoo-view-background-color;
     margin-bottom: 0px;
 
+    // Checkbox in editable list
+    // should be clickable and activate the row
+    &.o_editable_list .o_data_row:not(.o_selected_row) {
+        .o_checkbox:not(.o_readonly_modifier) {
+           pointer-events: none;
+        }
+    }
```

Comment:
this selector matches the raw selection checkboxes (at the beginning of the lines) for non x2many list, making them inert